Windows

rminnich opened the pull request:
yep, not kidding. This lets me work under windows. It's not all there, consider this a WIP until next week.

rminnich committed:
checkpoint windows support

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rjoleary commented on client/cpu9p_windows.go:
```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
You should mention Apache in the top-most LICENSE file -- some people might be surprised that some files are under different licenses.

rjoleary commented on client/cpu9p_windows.go:
```diff
@@ -0,0 +1,141 @@
+
+// GetAttr implements p9.File.GetAttr.
+//
+// Not fully implemented.
```
Mention one thing that's not implemented. Or is it not implemented because of Windows?

rminnich replied:
done

rjoleary commented on client/cpu9p_windows.go:
```diff
@@ -0,0 +1,141 @@
+		NLink:            3,
+		Size:             uint64(fi.Size()),
+		BlockSize:        uint64(512),
+		Blocks:           uint64(fi.Size() / 512),
```
I feel like this should round up: `(size + 511) / 512`

rjoleary commented on client/cpu9p_windows.go:
```diff
@@ -0,0 +1,141 @@
+		RDev:   true,
+		Size:   true,
+		Blocks: true,
+		ATime:  true,
```
It seems like `ATime` stands for "Access Time" and we're setting it to `ModTime` which is "Modified Time" which seems incorrect. Is it possible to just set this AttrMask value to false to indicate we don't know the `ATime`?

rjoleary commented on go.mod:
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/hugelgupf/p9 v0.2.1-0.20230814004337-e6037077d6dc
 	github.com/kevinburke/ssh_config v1.2.0
-	github.com/u-root/u-root v0.11.1-0.20230913033713-004977728a9d
+	github.com/u-root/u-root v1.0.1
```
Is this intentional? I thought we used Go-style versioning where it's risky to go above v0.x.

rjoleary reviewed (commented):
Windows support is a hug leap!

rminnich committed:
cpio9p: do not use path/filepath

This is a cpio archive, so we must use path.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

rminnich commented:
Thanks so much for that review! I will tidy things up tomorrow. Your suggestions are excellent.

rminnich commented:
actually, I can't move forward on this until the termios windows support on u-root is upstreamed, though I may fix that problem (the review is never happening?) and move to using some other termios package, as gosh seems to do.

rminnich committed:
checkpoint more windows support

we're currently stuck on termios, on u-root, and if that
does not get reviewed soon I'll move to using something else.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

rminnich committed:
client: use golang/x/term instead of u-root termios

u-root termios has much more capability than we need, and
does not work on windows yet.

Signed-off-by: Ron Minnich <rminnich@gmail.com>

rminnich committed:
client/client.go: windows sucks

you can make things raw, but sometimes, you can't get the
window size.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

rminnich committed:
windows:bring in the corrected getattr; remove attr_windows.go

Signed-off-by: Ron Minnich <rminnich@gmail.com>

rminnich committed:
cpu/client: fix simple error while testing windows term

Signed-off-by: Ron Minnich <rminnich@gmail.com>

rminnich commented:
idk about windows any more
